lib/src/utils/node_utils.dart

```diff
@@ -22,3 +24,66 @@ String humanReadableNodeType(AstNode? node) {
 
   return 'Node';
 }
+
+/// Parses the provided type string to extract a [NamedType].
+NamedType parseNamedTypeFromString(String typeString) {
+  try {
+    final namedTypeFinder = _NamedTypeFinder();
+
+    final parseResult = parseString(content: "$typeString _;");
+    parseResult.unit.visitChildren(namedTypeFinder);
+
+    return namedTypeFinder.foundNamedType!;
+  } catch (_) {
+    throw Exception("No NamedType could be parsed from the input "
+        "typeString: '$typeString'. Ensure it's a valid Dart "
+        "type declaration.");
+  }
+}
+
+class _NamedTypeFinder extends GeneralizingAstVisitor<void> {
+  NamedType? _foundNamedType;
+
+  NamedType? get foundNamedType => _foundNamedType;
+
+  @override
+  void visitNamedType(NamedType namedType) {
+    _foundNamedType ??= namedType;
+  }
+}
+
+///
+extension ChildNamedTypes on NamedType {
+  /// Retrieves child [NamedType] instances from type arguments.
+  List<NamedType> get childNamedTypes =>
+      typeArguments?.arguments.whereType<NamedType>().toList() ?? [];
+
+  /// Gets the token name of this type instance.
+  String get tokenName => name2.toString();
+
+  /// Checks if the current token name is 'dynamic'.
+  bool get isDynamic => tokenName == "dynamic";
+
+  /// Checks if the current token name is 'Object'.
+  bool get isObject => tokenName == "Object";
+
+  /// Checks if this node is a subtype of the specified node
+  /// based on their structures.
+  bool isSubtype({required NamedType node}) {
```

Comment:
I think this name communicates more clearly the direction of the comparison.

```suggestion
  bool isSubtypeOf({required NamedType node}) {
```